enrichments/internal/elastic/resource.go

```diff
@@ -174,14 +171,9 @@ func (s *resourceEnrichmentContext) overrideHostNameWithK8sNodeName(resource pco
 	)
 }
 
-// setServiceInstanceID sets service.instance.id from container.id or host.name
-// if service.instance.id is not already set. This follows the existing APM logic for
-// `service.node.name`.
+// setServiceInstanceID sets service.instance.id from container.id or host.name.
+// This follows the existing APM logic for `service.node.name`.
 func (s *resourceEnrichmentContext) setServiceInstanceID(resource pcommon.Resource) {
-	if s.serviceInstanceID != "" {
-		return
-	}
-
```

Comment:
Why is this removed now and an existing `s.serviceInstanceID` overwritten? If this is an intended change for apm logic compliance, it might need to go into a dedicated PR to not mix with the existing attribute handling.